### PR TITLE
Indicate bug step number in the left bug step overview tree

### LINKED ISSUE #778: Indicate bug step number in the left bug step overview tree
Indicate the bug step sequence number in the left tree form bug-step view so that it will be easier for a user to understand which bug step corresponds to which indicated note in the code.

Also highlight the currently clicked bug path step better (the current marking is almost invisible).

### PR DESCRIPTION
**Pull request by csordasmarton**
Closes #778

### CODE REVIEW

**Comment by csordasmarton** [comment]
I made a screenshot from this:
cc_bugstep

**Comment by Xazax-hun** [comment]
I do like how this looks, however, I am not sure if red is a good choice of color here. My problem is, red tends to be the color of errors. In this case, `Assuming...` is not an error.

**Comment by whisperity** [comment]
I feel that the numbers are too big, compared to the rest of the GUI.

**Comment by csordasmarton** [comment]
@Xazax-hun I separated the colors of the bubbles to blue (information) and red (error - last item). What do you think about this?

@whisperity The numbers are bigger, because I added a bold style to it. I removed it.

cc_bugstep_2

**Comment by Xazax-hun** [comment]
Looks good, thank you!

**Comment by whisperity** [comment]
`fixit`s contain a special [` (fixit)` marker](https://github.com/Ericsson/codechecker/blob/7d031bfa4241d277ac6b3e18116656fcb4dfa031/libcodechecker/analyze/tidy_output_converter.py#L337) at the end of their string. Maybe we should use a different highlight (yellow? orange? green?... I'd vouch for yellow.) for those?

Currently it looks like this:
image

@Xazax-hun, what do you think?

**Comment by Xazax-hun** [comment]
@whisperity ,I agree, good idea.

**Comment by whisperity** [comment]
As a final tweak what I would like to see, if possible, is **not** showing the bug number if the entire bug is consistent of a single line (e.g. `deadcode.DeadStores`).

Consider that if there aren't any steps, only one report is shown in the tree view (and that has no number), so I believe the marker in the code viewer shouldn't have a number marker either.

image

Other than this, this looks lovely!

File: www/scripts/codecheckerviewer/BugViewer.js [hunk 1]
```diff
@@ -78,6 +78,31 @@ function (declare, domClass, dom, style, fx, Toggler, on, query, Memory,
...
+
+  /**
+   * Parsing the bug step message and replace the `(fixit)` string with an icon.
+   */
+  function parseBugStepMsg(message) {
+    var ret = message;
+
+    if (message.indexOf('(fixit)') > -1)
+      ret = '<span class="tag fixit">fixit</span>'
+        + message.replace('(fixit)', '');
```
Comment (whisperity) [inline www/scripts/codecheckerviewer/BugViewer.js]: The text is appended as ` (fixit)` with a space before the tag. While this is a very minor stuff, I think we should stript in entirely.

**Pull request merged**